prusti-viper/src/encoder/spec_encoder.rs

```diff
@@ -387,6 +495,7 @@ impl<'p, 'v: 'p, 'tcx: 'v> SpecEncoder<'p, 'v, 'tcx> {
         replacements.extend(
             mir.args_iter()
                 .zip(self.target_args)
+                .skip(if skip_first { 1 } else { 0 })
```

Comment:
Is it correct to ignore the first element of `self.target_args` when `skip_first` is `true`? If not, should the `.skip(..)` be before the `.zip(..)`?

Reply:
This panics when the order is swapped, so I think it is correct like this. The `self` argument is also skipped in `spec_function_encoder` when creating `formal_args`.